sdks/python/apache_beam/transforms/sideinputs_test.py

```diff
@@ -309,6 +316,65 @@ def test_flattened_side_input(self):
     assert_that(results, equal_to(['a', 'b']))
     pipeline.run()
 
+  @attr('ValidatesRunner')
+  def test_multi_triggered_gbk_side_input(self):
+    """Test a GBK sideinput, with multiple triggering."""
+    options = PipelineOptions()
+    options.view_as(StandardOptions).streaming = True
+    p = TestPipeline(options=options)
+
+    main_stream = (p
+                   | 'main TestStream' >> TestStream()
+                   .advance_watermark_to(3)
+                   .add_elements(['a1'])
+                   .advance_watermark_to(8)
+                   .add_elements(['a2'])
+                   | 'main windowInto' >> beam.WindowInto(
+                       window.FixedWindows(5),
+                       trigger=trigger.AfterWatermark(
+                           early=trigger.AfterCount(1)),
+                       accumulation_mode=trigger.AccumulationMode.DISCARDING))
+
+    emit_vals = Map(lambda k_vs: k_vs[1])
+
+    side_stream = (p
+                   | 'side TestStream' >> TestStream()
+                   .add_elements([window.TimestampedValue(('k', 100), 2)])
+                   .add_elements([window.TimestampedValue(('k', 400), 7)])
+                   | 'side windowInto' >> beam.WindowInto(
+                       window.FixedWindows(5),
+                       trigger=trigger.AfterWatermark(
+                           early=trigger.AfterCount(1)),
+                       accumulation_mode=trigger.AccumulationMode.DISCARDING)
+                   | 'GBK' >> beam.GroupByKey()
+                   | 'Values' >> emit_vals)
+
+    class RecordFn(beam.DoFn):
+      def process(self,
+                  elm=beam.DoFn.ElementParam,
+                  ts=beam.DoFn.TimestampParam,
+                  side=beam.DoFn.SideInputParam):
+        yield (elm, ts, side)
+
+    records = (main_stream
+               | beam.ParDo(RecordFn(), beam.pvalue.AsList(side_stream)))
+
+    expected_window_to_elements = {
+        window.IntervalWindow(0, 5): [
+            ('a1', Timestamp(3), [[100]]),
+        ],
+        window.IntervalWindow(5, 10): [
+            ('a2', Timestamp(8), [[400], []])
```

Comment:
That empty list is odd...

Reply:
issue is gone after using multiple output test stream